src/types/index.ts

```diff
@@ -1284,6 +1284,13 @@ export interface ProcedureOpts {
    * Account or address of a signing key to replace the current one (for this procedure only)
    */
   signingAccount?: string | Account;
+
+  /**
+   * nonce value for signing the transaction
+   *
+   * More information can be found at: https://polkadot.js.org/docs/api/cookbook/tx/#how-do-i-take-the-pending-tx-pool-into-account-in-my-nonce
+   */
+  nonce?: BigNumber;
```

Comment:
We need to expose a method to get the nonce of an `Account`. I think a method like `currentNonce` there would be reasonable. I don't think there is a way to get that now, which makes this hard to use.

Then for this field we can link too that method instead about getting the current nonce for the signingAccount. 


Also, what do you think about having this be `BigNumber | () => BigNumber`  or even `() => BigNumber`. By giving a function a user could call `currentNonce` and make a closure and be able to increment it. All we'd have to do is call it and use the result.

For example:
```ts
  let nonce = 0
  const incNonce = (n) => () => (nonce += n);
  const addOne = incNonce(1);

  console.log(addOne());  // 1
  console.log(addOne()); // 2
  console.log(addOne()); // 3
  console.log(nonce); // 3
```

Reply:
Agree on everything. I would leave the option of passing a plain number as well as a function